collector-kinesis/src/main/java/zipkin/collector/kinesis/KinesisCollector.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    public static final class Builder implements CollectorComponent.Builder {
+
+        Collector.Builder delegate = Collector.builder(KinesisCollector.class);
+
+        AWSCredentialsProvider credentialsProvider;
+        String appName;
+        String streamName;
+
+        @Override
+        public Builder storage(StorageComponent storageComponent) {
+            delegate.storage(storageComponent);
+            return this;
+        }
+
+        @Override
+        public Builder metrics(CollectorMetrics collectorMetrics) {
+            delegate.metrics(Util.checkNotNull(collectorMetrics, "metrics").forTransport("kinesis"));
+            return this;
+        }
+
+        @Override
+        public Builder sampler(CollectorSampler collectorSampler) {
+            delegate.sampler(collectorSampler);
+            return this;
+        }
+
+        public Builder credentialsProvider(AWSCredentialsProvider credentialsProvider) {
+            this.credentialsProvider = credentialsProvider;
+            return this;
+        }
+
+        public Builder appName(String appName) {
+            this.appName = appName;
+            return this;
+        }
+
+        public Builder streamName(String streamName) {
+            this.streamName = streamName;
+            return this;
+        }
+
+        @Override
+        public KinesisCollector build() {
+            return new KinesisCollector(this);
+        }
+
+        Builder() {
+        }
+    }
+
+    private Collector collector;
+    private String appName;
+    private String streamName;
+    private AWSCredentialsProvider credentialsProvider;
+
+    private Executor executor = Executors.newSingleThreadExecutor();
+    private Worker worker;
+    private IRecordProcessorFactory processor;
+
+    KinesisCollector(Builder builder) {
+        this.collector = builder.delegate.build();
+
+        this.appName = builder.appName;
+        this.streamName = builder.streamName;
+        this.credentialsProvider = builder.credentialsProvider;
+    }
+
+    @Override
+    public KinesisCollector start() {
+        String workerId = null;
+        try {
+            workerId = InetAddress.getLocalHost().getCanonicalHostName() + ":" + UUID.randomUUID();
+        } catch (UnknownHostException e) {
+            workerId = UUID.randomUUID().toString();
+        }
+        KinesisClientLibConfiguration config = new KinesisClientLibConfiguration(appName, streamName, credentialsProvider, workerId);
+        processor = new KinesisRecordProcessorFactory(collector);
+        worker = new Worker.Builder()
+                .recordProcessorFactory(processor)
+                .config(config)
+                .build();
+
+        executor.execute(worker);
+        return this;
+    }
+
+    @Override
+    public CheckResult check() {
+        // TODO find a way to health check it
+        return CheckResult.OK;
+    }
+
+    @Override
+    public void close() throws IOException {
+        worker.shutdown();
```

Comment:
So because I am using a `newSingleThreadExecutor` there is no shutdown method. Once the worker loop finishes the thread will finish